sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -229,6 +229,7 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
     // List of SQL queries to run
     // note: this is not a robust way to split queries using semicolon, but works for now.
     val queries = code.mkString("\n").split("(?<=[^\\\\]);").map(_.trim).filter(_ != "").toSeq
+      .map(_.split("\n").filterNot(_.startsWith("--")).mkString("\n")).map(_.trim).filter(_ != "")
```

Comment:
I feel this is a little complicated, so could you describe what this code does in the comment?

Reply:
OK

Reply:
The description is a little misleading.
`The comment at the end of the query` looks like `SELECT 1 -- okay`.
Technically, it's split by line 231 (not line 233).

Given that, the following is a simpler solution, isn't it?
```
- val queries = code.mkString("\n").split("(?<=[^\\\\]);").map(_.trim).filter(_ != "").toSeq
+ val queries = code.mkString("\n").split("(?<=[^\\\\]);").map(_.trim).filter(_ != "").filterNot(_.startsWith("--")).toSeq
```

Reply:
(I didn't break the line in the above example.)

Reply:
While testing my idea, I found that we (this PR and my idea) still miss the following cases.
- https://github.com/apache/spark/blob/master/sql/core/src/test/resources/sql-tests/inputs/pgSQL/with.sql#L137-L141
```
CREATE TABLE department (
	id INTEGER,  -- department ID
	parent_department INTEGER, -- upper department ID
	name string -- department name
) USING parquet;
```

Technically, this PR is able to remove a single comment or a comment after `;`.
Also, the comments in the middle seems to be harmless.

Reply:
The current implementation turns out stronger. I'm trying to simplify this PR (with more elimination).